### PR TITLE
[MXNet-1002] Add GluonCV and NLP tookits, Keras, and developer wiki to navigation

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Description ##

* Add a link to the Developer Wiki under Docs
* Add links to Gluon Toolkits under Gluon
* Change name of Gluon > Tutorials to be "The Straight Dope (Tutorials)"
* Keras to API list
* Sync the mobile menu to match the desktop menu (for the most part). I left Install where it was, but removed tutorials and github because they are already in other submenus.

Fixes: https://issues.apache.org/jira/browse/MXNET-1002

## Preview
**UPDATE**: changing the preview link so I can continue to host the preview for search and redirects without disrupting this one. Note, this link will not work until the latest commit passes CI. I'm bumping the link to version 7 in anticipation.
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12704/7/tutorials/gluon/mnist.html

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot [pr-awaiting-review, Doc]

**Comment by anirudhacharya** [comment]
@aaronmarkham the preview is not working, it takes me to this page - https://mxnet.incubator.apache.org/error/404.html

**Comment by aaronmarkham** [comment]
@anirudhacharya - conflicting previews with my two outstanding PRs. The other one enforces redirects to the default version... I disabled that for now. Please check again.

**Comment by anirudhacharya** [comment]
@aaronmarkham just checked. I am still facing the same issue.

File: docs/_static/mxnet-theme/navbar.html [hunk 1]
```diff
@@ -22,6 +24,7 @@ <h1 id="logo-wrap">
             <li><a class="main-nav-link" href="{{url_root}}api/c++/index.html">C++</a></li>
             <li><a class="main-nav-link" href="{{url_root}}api/clojure/index.html">Clojure</a></li>
             <li><a class="main-nav-link" href="{{url_root}}api/julia/index.html">Julia</a></li>
+            <li><a class="main-nav-link" href="https://github.com/awslabs/keras-apache-mxnet">Keras</a></li>
```
Comment (szha) [inline docs/_static/mxnet-theme/navbar.html]: keras is not a language binding. It's a community project and should reside somewhere else.

**Comment by aaronmarkham** [comment]
@anirudhacharya Please clear your cache. I've noticed that once you get a redirect, that can be cached and you won't get the new page without clearing it.

@szha I noticed that people search for it on the site, but really nothing comes up. It should be highlighted. Keras-MXNet is sort of a meta-binding, in that you write another language for your network, but use MXNet. Seems like a good spot to me. Just having it on the ecosystem page isn't enough in my opinion. Also, consider the Perl API. It is maintained as a community project with the docs elsewhere. Isn't that pretty similar?

**Comment by smolix** [comment]
@szha - I take the blame for this suggested modification. Keras support is a key feature that we have in MxNet. Hence we need a place to expose this prominently. I know that it's not really a language binding but there's no other place where it would fit even approximately. 

Happy to put it somewhere else where everyone can find out about it easily. But I think we're underselling ourselves by not mentioning it.

**Comment by szha** [comment]
@smolix I'd suggest a separate page like https://www.tensorflow.org/ecosystem/
image

**Comment by szha** [comment]
In comparison, this is what tensorflow has in API:
image

I'm fine with advertising good community projects. I'm not fine with having taxonomical error on the main page. As long as we have a good and right place to advertise it, I'm happy. Coupling the API documentation (& version) will complicate the development cycles later.

**Comment by aaronmarkham** [comment]
There's a link to Keras-MXNet on our ecosystem page. However, this page gets very little traffic.
One week visits: 48
% of Total: 0.14% (34,971)
It appears that nothing is coming in from google search, and some small amount is coming from people coming to the site then using the mxnet site to search for "keras" and then finding the ecosystem page. I noted that mxboard is also searched for and we have very little to show of that project on the site.
If ecosystem were brought out to be a top level navigation (like the tf site) and highlighted projects like keras were second level, then maybe we'd get a big boost for it. 
With regard to the versions, the way the site is working now, it won't be an issue as this kind of content would be common for every version (and keras links off-site). I'm redirecting traffic for old versions that don't have the new APIs to the API error page (suggesting a newer version).

**Comment by szha** [comment]
@aaronmarkham can we make it more prominent? Notice that tensorflow has separate community and ecosystem dropdowns

**Comment by szha** [comment]
Right now that page is hiding here: 
image

**Comment by aaronmarkham** [comment]
@szha So I'm removing Keras for now, so we can get the benefit of the other updates in this PR.
I think adding a new menu top level navigation may cause more debate and further tie up any release, so I feel that I should propose that change in a future PR.

**Comment by szha** [comment]
OK. I think with the same amount of space we have better ways of organizing the documents. Right now the tab "Doc" is a mixture of things, some of which belong in the category of "getting started" while some are of the category "development". Feel free to propose a layout that helps discovery

**Comment by szha** [comment]
Preview at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-12704/8/index.html. Looks good to me.
